### PR TITLE
needs to use {{this}} format for incremental

### PR DESCRIPTION
**Pull request by andrewhong5297**
Brief comments on the purpose of your changes:

We were getting duplicates with this always running with one day of history, I've changed it to only run if the block_time is in the next day.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by andrewhong5297** [comment]
```
Expected a schema version of "https://schemas.getdbt.com/dbt/manifest/v6.json" in manifest.json, but found "https://schemas.getdbt.com/dbt/manifest/v8.json". Are you running with a different version of dbt?
```

hmm, seems like something is wrong with config? cc @jeff-dude

**Review by Hosuke — APPROVED** [review]
{{this}} provides a more precise incremental updating option.
Original idea is having an overlapping region when incremental update, to ensure 100% coverage of new incoming records, and dbt uses `unique_key` in the config block to filter out the duplicated/existed records when updating.

File: models/solana_utils/solana_utils_daily_balances.sql [hunk 1]
```diff
@@ -24,7 +24,7 @@ WITH
             FROM {{ source('solana','account_activity') }}
             WHERE tx_success 
             {% if is_incremental() %}
-            AND block_time >= date_trunc("day", now() - interval '1 day')
+            AND block_time > (SELECT max(day) FROM {{this}})
```
Comment (0xRobin) [inline models/solana_utils/solana_utils_daily_balances.sql:27]: should probably use `>=` here to not drop later incremental updates from the same day.

**Comment by 0xRobin** [comment]
CI issues should be resolved now, I've triggered a re-run

**Comment by jeff-dude** [comment]
can you confirm the failing test on duplicates?

**Comment by andrewhong5297** [comment]
> can you confirm the failing test on duplicates?

Looking

**Comment by andrewhong5297** [comment]
the test schema is gone, but I really can't see any reason there would be duplicates given we're using a window function.

**Comment by andrewhong5297** [comment]
@jeff-dude I'm like 100% confident there should be no duplicates - can we merge?

**Comment by Hosuke** [comment]
I think I know the cause of the bug,
there is a tiny time diff in `updated_at`(1st time in full-refresh, 2nd time in incremental refresh).
Perhaps the engine is hard to compare NULL value for `token_mint_address`.
Here is an example:
https://dune.com/queries/2493263?d=1

**Comment by Hosuke** [comment]
I tried to fix the problem by giving a default empty string `''` to NULL `token_mint_address`, and currently empty string is not equal to NULL in some versions of spark.

I actually found out this problem is an opening issue in dbt: https://github.com/dbt-labs/dbt-core/issues/7597

**Comment by andrewhong5297** [comment]
hmm odd. I've run a group by check and got no nulls https://dune.com/queries/2479876?d=1&q=v2+&sidebar=query-explorer

**Comment by andrewhong5297** [comment]
thank you @Hosuke !!

**Comment by jeff-dude** [comment]
> I tried to fix the problem by giving a default empty string `''` to NULL `token_mint_address`, and currently empty string is not equal to NULL in some versions of spark.
> 
> I actually found out this problem is an opening issue in dbt: [dbt-labs/dbt-core#7597](https://github.com/dbt-labs/dbt-core/issues/7597)

great catch! we've seen this elsewhere before too. it's a good thing to keep in mind, that unique keys can't be null or the merge statement fails the join condition, leading to duplicate inserts. i wonder if this means we need to apply `not null` tests to each unique key column 🤔

**Comment by jeff-dude** [comment]
seems like we're getting failures where not null tests were applied (assuming the unique keys)

**Comment by andrewhong5297** [comment]
> seems like we're getting failures where not null tests were applied (assuming the unique keys)

Oh token_mint_address and token_balance can be null, those tests should he removed. Just address and balance should have not null tests.

**Comment by jeff-dude** [comment]
what's the move here?

**Comment by jeff-dude** [comment]
closing PRs marked as `wip` state, prior to the freeze announced and tracked [here](https://github.com/duneanalytics/spellbook/discussions/3558)

**Pull request closed without merging**